server/routes/auth/providers/github.js

```diff
@@ -0,0 +1,115 @@
+// @flow
+import passport from "@outlinewiki/koa-passport";
+import fetch from "fetch-with-proxy";
+import Router from "koa-router";
+import { Strategy as GithubStrategy } from "passport-github2";
+import accountProvisioner from "../../../commands/accountProvisioner";
+import env from "../../../env";
+import passportMiddleware from "../../../middlewares/passport";
+import { StateStore } from "../../../utils/passport";
+
+const router = new Router();
+const providerName = "github";
+const GITHUB_CLIENT_ID = process.env.GITHUB_CLIENT_ID;
+const GITHUB_CLIENT_SECRET = process.env.GITHUB_CLIENT_SECRET;
+const GITHUB_ORGANIZATION = process.env.GITHUB_ORGANIZATION;
+
+const scopes = ["read:org", "user:email"];
+
+export async function request(endpoint: string, accessToken: string) {
+  const response = await fetch(endpoint, {
+    method: "GET",
+    headers: {
+      Authorization: `token ${accessToken}`,
+      "Content-Type": "application/json",
+    },
+  });
+
+  if (response.status === 204) {
+    return {};
+  }
+
+  return response.json();
+}
+
+export const config = {
+  name: "Github",
+  enabled: !!GITHUB_CLIENT_ID,
+};
+
+if (GITHUB_CLIENT_ID) {
+  if (GITHUB_ORGANIZATION === "") {
+    throw new Error("`GITHUB_ORGANIZATION` required when using Github.");
+  }
+
+  const strategy = new GithubStrategy(
+    {
+      clientID: GITHUB_CLIENT_ID,
+      clientSecret: GITHUB_CLIENT_SECRET,
+      callbackURL: `${env.URL}/auth/github.callback`,
+      passReqToCallback: true,
+      store: new StateStore(),
+      scope: scopes,
+    },
+    async function (req, accessToken, refreshToken, profile, done) {
+      try {
+        const email = profile.emails[0].value;
+
+        const orgInfo = await request(
+          `https://api.github.com/orgs/${GITHUB_ORGANIZATION}`,
+          accessToken
+        );
+
+        if (orgInfo.message === "Not Found") {
+          throw new Error("Failed to retrieve organization information.");
+        }
+
+        // Verify if you are member of this orgnaization
+        const isMember = await request(
+          `https://api.github.com/orgs/${GITHUB_ORGANIZATION}/members/${profile.username}`,
+          accessToken
+        );
+
+        if (isMember.message === "Not Found") {
+          throw new Error("User not a member of the organization.");
+        }
+
+        const result = await accountProvisioner({
+          ip: req.ip,
+          team: {
+            name: orgInfo.name,
```

Comment:
```suggestion
            name: orgInfo.company || orgInfo.name,
```

Use the proper company name if available?